src/client/Microsoft.Identity.Client/Kerberos/Win32/LsaTokenSafeHandle.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using static Microsoft.Identity.Client.Kerberos.Win32.NativeMethods;
+
+namespace Microsoft.Identity.Client.Kerberos.Win32
+{
+    internal class LsaTokenSafeHandle : SafeHandle
+    {
+        public LsaTokenSafeHandle()
+            : base(IntPtr.Zero, true)
+        {
+        }
+
+        public bool Impersonating { get; private set; }
+
+        public override bool IsInvalid => this.handle == IntPtr.Zero;
+
+        protected override bool ReleaseHandle()
+        {
+            this.Revert();
+
+            if (!CloseHandle(this.handle))
+            {
+                var error = Marshal.GetLastWin32Error();
+
+                throw new Win32Exception(error);
+            }
+
+            return true;
+        }
+
+        public void Impersonate()
+        {
```

Comment:
Is this needed? Seems like it isnt used anywhere?

Reply:
This code part just copied from Kerberos.NET public package (https://www.nuget.org/packages/Kerberos.NET/) to support Windows Ticket Cache interface. I want to keep the code as it is.